render/wgpu/shaders/gradient.wgsl

```diff
@@ -16,32 +16,22 @@ struct VertexOutput {
 #endif
 
 struct Gradient {
-    colors: array<vec4<f32>, 16>,
-    ratios: array<vec4<f32>, 4u>, // secretly array<f32; 16> but this let's us squeeze it into alignment
-    gradient_type: i32,
-    num_colors: u32,
-    interpolation: i32,
     focal_point: f32,
+    interpolation: i32,
+    shape: i32,
+    repeat: i32,
 };
 
 #if use_push_constants == true
     @group(1) @binding(1) var<uniform> gradient: Gradient;
+    @group(1) @binding(2) var texture: texture_2d<f32>;
+    @group(1) @binding(3) var texture_sampler: sampler;
 #else
     @group(3) @binding(1) var<uniform> gradient: Gradient;
+    @group(3) @binding(2) var texture: texture_2d<f32>;
+    @group(3) @binding(3) var texture_sampler: sampler;
 #endif
 
-fn ratio(i: u32) -> f32 {
-    return gradient.ratios[i / 4u][i % 4u];
-}
-
-fn find_t(focal_point: f32, uv: vec2<f32>) -> f32 {
-    return 0.0;
-}
-
-/// FIXME: We should import VertexInput from 'common', but a naga_oil bug prevents
-/// us from importing 'common' in both this file and the invidual gradient shaders (like 'linear.wgsl')
-/// Currently, importing 'common' in that way will cause all of the definitions from 'common' to be duplicated,
-/// result in a WGSL parse error.
```

Comment:
Is this comment really not relevant anymore?

Reply:
Not strictly, after we split up the one global VertexInput into different types